qingstor/sdk/client/encryption_client.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import unicode_literals
+
+import base64
+import hashlib
+
+from Crypto.Cipher import AES
+from ..utils.helper import md5_digest
+
+from qingstor.sdk.service.bucket import Bucket
+
+class EncryptionClient(Bucket):
+
+    def __init__(self,config,client,bucket_name,zone,encrypt_key,encrypt_algo="AES256"):
+        properties = {"bucket-name": bucket_name, "zone": zone}
+        Bucket.__init__(self,config,properties,client)
+        self.encrypt_algo=encrypt_algo
+        self.encoded_encrypt_key = base64.b64encode(encrypt_key)
+        self.encoded_encrypt_key_md5 = base64.b64encode(md5_digest(encrypt_key))
+
+    def put_object(self,*args,**kwargs):
+        kwargs=self.apply_encrypt_headers(**kwargs)
+        return Bucket.put_object(self,*args, **kwargs)
+
+    def get_object(self,*args,**kwargs):
+        kwargs = self.apply_encrypt_headers(**kwargs)
+        return Bucket.get_object(self,*args, **kwargs)
+
+    def put_object_copy(self,*args,**kwargs):
+        kwargs = self.apply_copy_encrypt_headers(**kwargs)
+        return Bucket.put_object(self,*args, **kwargs)
+
+    def put_object_move(self,*args,**kwargs):
+        kwargs = self.apply_encrypt_headers(**kwargs)
+        return Bucket.put_object_move(self,*args, **kwargs)
+
+    def head_object(self,*args,**kwargs):
+        kwargs = self.apply_encrypt_headers(**kwargs)
+        return Bucket.head_object(self,*args, **kwargs)
+
+    def initiate_multipart_upload(self,*args,**kwargs):
+        kwargs = self.apply_encrypt_headers(**kwargs)
+        return Bucket.initiate_multipart_upload(self,*args, **kwargs)
+
+    def upload_multipart(self,*args,**kwargs):
+        kwargs = self.apply_encrypt_headers(**kwargs)
+        return Bucket.bucket.upload_multipart(self,*args, **kwargs)
+
+    def upload_multipart_copy(self,*args,**kwargs):
```

Comment:
copy and move don't need to implement because `copy_source` should passed by user.